lib/rexml/source.rb

```diff
@@ -226,8 +226,11 @@ def read(term = nil, min_bytes = 1)
     end
 
     def read_until(term)
-      pattern = Private::PRE_DEFINED_TERM_PATTERNS[term] || /#{Regexp.escape(term)}/
-      term = encode(term)
+      pattern = Private::PRE_DEFINED_TERM_PATTERNS[term]
+      if pattern.nil?
+        pattern = /#{Regexp.escape(term)}/
+        term = encode(term)
```

Comment:
Does this work when encoding is `UTF-16` or `UTF-32`?

Reply:
Sorry.
I have confirmed that there is a problem with `UTF-16` and will change it to draft.